pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -145,6 +153,17 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	if shouldRetry, err := r.checkTLSConfig(mdb); err != nil {
+		if shouldRetry {
```

Comment:
I don't understand this: the pattern is if err is not nil, the value returned should not be used